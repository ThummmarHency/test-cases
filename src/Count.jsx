import React, { useState } from "react";
import Table from "./component/Table";

const Count = () => {
  const [count, setCount] = useState(0);
  const handleCount = () => {
    setCount(count + 1);
  };
  return (
    <>
      <div className={`count-${count}`}>{count}</div>
      <button onClick={handleCount}>Click</button>
      <Table/>

    </>
  );
};

export default Count;
