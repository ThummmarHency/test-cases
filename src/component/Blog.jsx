import React, { useState } from "react";

const Blog = () => {
  const [temp, setTemp] = useState({});
  console.log('temp :>> ', temp);
  return (
    <div className="mainCls">
      <ul>
        <li>React</li>
        <li key="Node" className="liCls">Node</li>
        <li className="liCls">Angular</li>
      </ul>
      <div className="number">
        <span>one</span>
        <span>two</span>
      </div>
      <div>
        <button className="my-button disable btn-primary">Submit</button>
      </div>
      <div className="some-class">Test</div>
    </div>
  );
};

export default Blog;
