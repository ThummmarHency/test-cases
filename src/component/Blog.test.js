import React from "react";
import { shallow } from "enzyme";
import Blog from "./blog";
import Count from '../Count';

describe.skip("Blog page testing", () => {
  const blogInstance = shallow(<Blog />);

  it("check class", () => {
    expect(blogInstance.exists(".mainCls")).toEqual(true);
  });
  // it('check state',()=>{
  //     console.log('blogInstance.debug() :>> ', blogInstance.debug());
  //     expect(blogInstance.state('temp')).toEqual(true)
  // })
  it("check html contain", () => {
    expect(blogInstance.contains(<li>React</li>)).toEqual(true);
  });
  it("check with all without classname", () => {
    expect(
      blogInstance.containsAllMatchingElements([
        <div>Test</div>, //true
        <span>one</span>, //true
      ])
    ).toEqual(true);
  });
  it("check with any without classname", () => {
    expect(
      blogInstance.containsAnyMatchingElements([
        <div>Tes</div>, //false
        <span>one</span>, //true
      ])
    ).toEqual(true);
  });
  it("find check", () => {
    expect(blogInstance.find(".liCls").length).toBe(2);
  });
  it("at check", () => {
    expect(blogInstance.find(".liCls").at(0).key()).toBe("Node");
  });
  it("child check", () => {
    expect(blogInstance.find(".number").childAt(0).type()).toBe("span");
  });
  it("children length check", () => {
    expect(blogInstance.find("ul").children().length).toBe(3);
  });
  it("has class check", () => {
    expect(blogInstance.find(".my-button").hasClass('disable')).toBe(true);
  });
  it("is class check", () => {
    expect(blogInstance.is('.mainCls')).toBe(true);
  });
});

describe.skip('shallow vs mount',()=>{
  const blogInstanceShallow = shallow(<Count />);
  // const blogInstanceMount = mount(<Count />);

  // console.log('blogInstanceShallow'.blogInstanceShallow.debug())
  it("shallow",()=>{
    // console.log('blogInstanceMount>>',blogInstanceMount.debug())
  })
})