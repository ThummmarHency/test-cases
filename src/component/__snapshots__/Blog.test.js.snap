// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow vs mount shallow 1`] = `ShallowWrapper {}`;
